src/nikhilMJun20/assignment27/MeetingRoom.java

```diff
@@ -0,0 +1,106 @@
+		}else {
+			System.out.println("Room Nalanda is already booked for "+nalandaBookingDuration+" hr(s).");
+			displayAvilableMeetingRoomsForBooking();
+		}
+	}
+	
+	void bookMeetingRoom(MEETINGROOM meetingRoomName) {
+		bookMeetingRoom(meetingRoomName, 1.0);
+	}
+	
+	void bookMeetingRoom(MEETINGROOM meetingRoomName, double bookingDurationHrs) {
+		if(meetingRoomName.equals(MEETINGROOM.NALANDA)) {
+			if(isNalandaBooked == false) {
+				isNalandaBooked = true;
+				nalandaBookingDuration = bookingDurationHrs;
+				System.out.println("Room Nalanda booked for "+nalandaBookingDuration+" hr(s).");
+			}else {
+				System.out.println("Room Nalanda is already booked for "+nalandaBookingDuration+" hr(s).");
+				displayAvilableMeetingRoomsForBooking();
+			}
+		}else if(meetingRoomName.equals(MEETINGROOM.TAKSHASHILA)) {
+			if(isTakshashilaBooked == false) {
+				isTakshashilaBooked = true;
+				takshashilaBookingDuration = bookingDurationHrs;
+				System.out.println("Room Takshashila booked for "+takshashilaBookingDuration+" hr(s).");
+			}else {
+				System.out.println("Room Takshashila is already booked for "+takshashilaBookingDuration+" hr(s).");
+				displayAvilableMeetingRoomsForBooking();
+			}
+		}else if(meetingRoomName.equals(MEETINGROOM.AGRAFORT)) {
+			if(isAgraFortBooked == false) {
+				isAgraFortBooked = true;
+				agraFortBookingDuration = bookingDurationHrs;
+				System.out.println("Room AgraFort booked for "+agraFortBookingDuration+" hr(s).");
+			}else {
+				System.out.println("Room AgraFort is already booked for "+agraFortBookingDuration+" hr(s).");
+				displayAvilableMeetingRoomsForBooking();
+			}
+		}else if(meetingRoomName.equals(MEETINGROOM.PRATPGADH)) {
+			if(isPratapGadhBooked == false) {
+				isPratapGadhBooked = true;
+				pratapGadhBookingDuration = bookingDurationHrs;
+				System.out.println("Room PratapGadh booked for "+pratapGadhBookingDuration+" hr(s).");
+			}else {
+				System.out.println("Room PratapGadh is already booked for "+pratapGadhBookingDuration+" hr(s).");
+				displayAvilableMeetingRoomsForBooking();
+			}
+		}
+	}
+	
+	static void displayInfo() {
+		System.out.println("Information of Booked Meeting Rooms:-");
+		if(isNalandaBooked)
+			System.out.println("Nalanda : "+nalandaBookingDuration+" hr(s).");
+		if(isTakshashilaBooked)
+			System.out.println("Takshashila : "+takshashilaBookingDuration+" hr(s).");
+		if(isAgraFortBooked)
+			System.out.println("AgraFort : "+agraFortBookingDuration+" hr(s).");
+		if(isPratapGadhBooked)
+			System.out.println("PratapGadh : "+pratapGadhBookingDuration+" hr(s).");
+	}
+	
+	static void displayAvilableMeetingRoomsForBooking() {
+		System.out.println("Currently Available Meeting Rooms:-");
+		if(!isNalandaBooked)
+			System.out.println("Nalanda");
+		if(!isTakshashilaBooked)
+			System.out.println("Takshashila");
+		if(!isAgraFortBooked)
+			System.out.println("AgraFort");
+		if(!isPratapGadhBooked)
+			System.out.println("PratapGadh");
```

Comment:
--Just a suggestion you can also write this condition in case all the rooms are booked and we call this method:
if(bookingStatusForNalanda && bookingStatusForTakshashila && bookingStatusForAgraFort && bookingStatusForPratapGadh)
			System.out.println("We are fully booked.");

Reply:
Thanks you, @VaishnaviVJun20!
This is a valuable suggestion.
I have implemented it.